sklearn/compose/_column_transformer.py

```diff
@@ -417,6 +422,20 @@ def _validate_output(self, result):
                     "The output of the '{0}' transformer should be 2D (scipy "
                     "matrix, array, or pandas DataFrame).".format(name))
 
+    def _index_output(self, Xs):
+        """
+        Record which transformer produced which column.
+        """
+        idx_ = 0
+        self.transformers_output_ = {}
+        for idx, (name, _, _, _) in enumerate(
```

Comment:
having `idx` and `idx_` is a bit confusing so I'd suggest naming this one more explicitly with `transformer_idx`

We can then rename `idx_` into `idx`, or `output_start_idx` if everything fits under the width constraint (not worth it otherwise)